### ISSUE TITLE
Java 8 + Serenity 1.1.36 = Hang + "can't receive further commands"

### ISSUE THREAD

**Issue by rrussell39** [serenity-bdd/serenity-core#456]
I recently began the work of migrating our automation framework from java 7 to Java 8.  Since there is so much testing involved we decided to update Serenity (from 1.1.29-rc.1 to 1.1.36) and other dependent libraries.
When I run my suite of tests locally using Firefox 47, they run great.  I even use surefire.forkNumber = 15.
However, when I run the tests on Saucelabs (various OS + Browser configs) I see all sorts of new failures including a repeatable hang towards the end of the test run.
Many, but not all, of the failures are "ERROR The test with session id ... has already finished, and can't receive further commands."
I call getWebDriver().quit() in my tear down process after each test such that every test gets a fresh browser instance. (I did see another issue posted here on GitHub related to this.)
Specs:
Before: Java 7, Serenity 1.1.29-rc.1, Selenium 2.52.0
After: Java 8, Serenity 1.1.36, Selenium 2.53.0
Junit = 4.12
Retry attempts = 1
Surefire is forking at the class level
I also updated several dependent libraries in terms of most up to date minor updates, no major updates. 
Any ideas or is anyone else experiencing pain migrating to Java 8 and/or Serenity 1.1.36?

**Comment by heerqa** [comment]
I was also facing issue when i updated to java 8 in codeship. But I fixed the issue by switching to java 8.
Not sure if these commands will work in Sauce lab
jdk_switcher home oraclejdk8
# /usr/lib/jvm/java-8-oracle
jdk_switcher use oraclejdk8
echo $JAVA_HOME
# /usr/lib/jvm/java-8-oracle
echo $JRE_HOME
# /usr/lib/jvm/java-8-oracle/jre
java -version

**Comment by rrussell39** [comment]
One other thing to note (as further testing completes).  When I update all my dependent libraries including Java 8 JDK but keep Serenity at 1.1.29-rc.1, things run fine.  It appears the issue I'm running into does not hinge on the Java version but on the Serenity/Selenium update.

**Comment by rrussell39** [comment]
@heerqa Thanks for your reply.  I don't manually control anything about the Saucelabs environment.  I simply pass my credentials and the os + browser config I want to test.  But see my previous post - it appears I'm having some conflict w/ the Serenity changes since 1.1.29

**Comment by wakaleo** [comment]
Why are you calling WebDriver.quit() manually? serenity should do this for you.

**Comment by rrussell39** [comment]
@wakaleo This part of the code has been around nearly two years so my hunch is that when we didn't close it originally, the browser was recycled.  I will try not calling quit() explicitly.

**Comment by wakaleo** [comment]
I'll take a look too. If you can isolate the version after 1.1.29 where the issue started happening (including the RC releases) it would help.

**Comment by wakaleo** [comment]
Is this issue appearing in version 1.2.2-rc.6?

**Comment by rrussell39** [comment]
I'm scheduled to do some Serenity update experimentation this sprint so I'll be in touch hopefully soon on some feedback.

**Comment by rrussell39** [comment]
Testing now.

**Comment by rrussell39** [comment]
Things are looking pretty good (using 1.2.2-rc.6).
1) Seems like things are categorized as success/failure/pending/ignored more than before
2) I have six tests now (out of 233) that are marked as 'ignored' but it appears it ran all steps successfully
I'll be researching things more and will report back what I find but wanted to give some early feedback.

**Comment by rrussell39** [comment]
I have not seen this issue using 1.2.2-rc.9.  I think this issue can be closed/fixed.
